nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,1825 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "nlp_gpt3mix.ipynb",
+      "provenance": [],
+      "collapsed_sections": []
+    },
+    "interpreter": {
+      "hash": "b3ba2566441a7c06988d0923437866b63cedc61552a5af99d1f4fb67d367b25f"
+    },
+    "kernelspec": {
+      "display_name": "Python 3",
+      "language": "python",
+      "name": "python3"
+    },
+    "language_info": {
+      "codemirror_mode": {
+        "name": "ipython",
+        "version": 3
+      },
+      "file_extension": ".py",
+      "mimetype": "text/x-python",
+      "name": "python",
+      "nbconvert_exporter": "python",
+      "pygments_lexer": "ipython3",
+      "version": "3.8.6"
+    },
+    "accelerator": "GPU"
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LXro9efVU84k"
+      },
+      "source": [
+        "# 📈: Text Augmentation using large-scale LMs and prompt engineering\n",
+        "\n",
+        "This notebook looks into the possibility of performing data augmentation on an NLP dataset leveraging the few-shot capabilities of large-scale language models (LMs) and prompt engineering 💡.\n",
+        "\n",
+        "Data augmentation techniques are used to generate additional samples. Data augmentation is already standard practice in computer vision projects 👀, but can also be leveraged in many NLP problems. We'll use a limited training set to simulate a real-world use case, where we are often constrained by the size of the available data 🤦."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "acwqGIAyU84m"
+      },
+      "source": [
+        "## 🛠️ Getting started\n",
+        "\n",
+        "The cells below will setup everything that is required to get started with data augmentation and finetuning an NLP model with the HuggingFace API."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "buIDjl_wU84m"
+      },
+      "source": [
+        "### Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "v-ItvjqP4Cxn"
+      },
+      "source": [
+        "!pip install -q transformers==4.12.5 datasets==1.16.1 tokenizers==0.10.3 openai==0.11.3 requests==2.23.0 sentencepiece==0.1.96"
```

Comment:
Add a piece on `pandas==1.1.0` as well, other versions (installed with openai) gave me errors further down

Reply:
The current way of working doesn't work:
`!pip install -q transformers==4.12.5 datasets==1.16.1 tokenizers==0.10.3 openai==0.11.3 requests==2.23.0 sentencepiece==0.1.96 pandas==1.1.0` gives me an error that it cannot install datasets, resulting in an importerror on the next cell:

![image](https://user-images.githubusercontent.com/22496343/146012585-b4ec05f0-70f7-4664-9412-91f7dc420211.png)

Reply:
When debugging these kinds of errors, it's always best to factory reset your notebook and test again